core/kernel/ldelf_loader.c

```diff
@@ -452,5 +490,12 @@ TEE_Result ldelf_dlsym(struct user_mode_ctx *uctx, TEE_UUID *uuid,
 			*val = arg->dlsym.val;
 	}
 
+out_free_dl_entry_arg:
+	if (dl_entry_arg)
+		free(dl_entry_arg);
+out:
+	if (uuid_kbuf)
+		free(uuid_kbuf);
```

Comment:
no need to test `dl_entry_arg` and `uuid_kbuf`. It is allowed to call `free(NULL)`.